### PR TITLE
Fix balance update logic and fallbacks

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes RNBW-1980  ( randomly showing ETH balance = 0 )

## What changed (plus any additional context for devs)
- Use covalent assets over mainnet assets (which have no balance / price)
- Prevent assets state updates when balance is missing on mainnet
- Prevent assets state updates when balance is missing on optimism

## PoW (screenshots / screen recordings)

Before: https://recordit.co/XvyI9hL2p3
After: http://recordit.co/UR6W9YiBqH

## Dev checklist for QA: what to test
- Pull to refresh / assets updating after sending txs
- set forceFallbackProvider = true and confirm asset balances work fine.

## Final checklist
[x] Assigned individual reviewers?
[x] Added labels?

### CODE REVIEW

**Review by derHowie — APPROVED** [review]
👍

File: src/redux/fallbackExplorer.js [hunk 1]
```diff
@@ -61,14 +62,16 @@ const getMainnetAssetsFromCovalent = async (
...
+      const isETH = toLower(contractAddress) === toLower(COVALENT_ETH_ADDRESS);
+      if (isETH) {
         contractAddress = ETH_ADDRESS;
       }
 
       const coingeckoId = coingeckoIds[toLower(contractAddress)];
       let price = {
         changed_at: updatedAt,
         relative_change_24h: 0,
+        value: isETH ? item.quote_rate : 0,
```
Comment (brunobar79) [inline src/redux/fallbackExplorer.js:74]: I trust covalent's price for ETH for not the other ones!

File: src/redux/fallbackExplorer.js [hunk 2]
```diff
@@ -312,7 +316,7 @@ export const fetchOnchainBalances = ({
     network === NetworkTypes.mainnet
       ? covalentMainnetAssets
         ? uniqBy(
-            [...mainnetAssets, ...covalentMainnetAssets],
+            [...covalentMainnetAssets, ...mainnetAssets],
```
Comment (brunobar79) [inline src/redux/fallbackExplorer.js:319]: this fixes the exclusion of covalent's updates

File: src/redux/fallbackExplorer.js [hunk 3]
```diff
@@ -407,7 +411,7 @@ export const fetchOnchainBalances = ({
 
   const newPayload = { assets: updatedAssets };
 
-  if (!keepPolling || !isEqual(lastUpdatePayload, newPayload)) {
+  if (balances && (!keepPolling || !isEqual(lastUpdatePayload, newPayload))) {
```
Comment (brunobar79) [inline src/redux/fallbackExplorer.js:414]: this prevents updates with missing balance

**Review by skylarbarrera — APPROVED** [review]
code lgtm

**Review by jinchung — APPROVED** [review]
🍔

**Pull request merged**